src/main/java/com/conveyal/r5/streets/EgressCostTable.java

```diff
@@ -269,96 +307,38 @@ public EgressCostTable (LinkedPointSet linkedPointSet, ProgressListener progress
-                    }
                 } else {
                     throw new UnsupportedOperationException("Tried to link a pointset with an unsupported street mode");
                 }
+                sr.quantityToMinimize = linkageCostUnit;
+                sr.route();
+                return linkedPointSet.extendCostsToPoints(sr.getReachedVertices()::get,
+                        sr.quantityToMinimize,
+                        envelopeAroundStop,
+                        egressArea);
             }
         }).collect(Collectors.toList());
         computeCounter.done();
         copyCounter.done();
-
-        // Now that we have a full set of tables, filter and transform them to reflect on-demand egress service.
-        // TODO optimization: we could return null immediately above for all stops not served by the on-demand egress:
-        // if (onDemandEgressService != null && ! onDemandEgressService.allows(stopIndex)) return null;
-        // We could also integrate the geographic filtering into that same loop and extendDistanceTableToPoints.
-        // In fact this is a major optimization since building car tables is very slow, and we might only need a few.
-        StreetLayer streetLayer = transitLayer.parentNetwork.streetLayer;
-        PickupWaitTimes pickupWaitTimes = streetLayer.pickupWaitTimes;
-        if (pickupWaitTimes != null && pickupWaitTimes.streetMode == streetMode) {
-            if (streetMode != StreetMode.CAR) {
-                // FIXME only cars have egress cost tables in seconds. Others will need a constant time offset field.
-                throw new RuntimeException("Only car egress tables can have a baked in time delay.");
-            }
-            for (int s = 0; s < stopToPointLinkageCostTables.size(); s++) {
-                PickupWaitTimes.EgressService egressService = pickupWaitTimes.getEgressService(s);
-                if (egressService == null || egressService.waitTimeSeconds < 0) {
-                    stopToPointLinkageCostTables.set(s, null);
-                    continue;
-                }
-                int[] costs = stopToPointLinkageCostTables.get(s);
-                TIntList filteredCosts = new TIntArrayList();
-                for (int i = 0; i < costs.length; i += 2) {
-                    int point = costs[i];
-                    int cost = costs[i + 1];
-                    // TODO normalize variable names (to costs?), these are not just times they may be distances.
-                    // TODO linkedPointSet.pointSet.getPointsInGeometry(), and pointInsideGeometry? default defs.
-                    double lat = linkedPointSet.pointSet.getLat(point);
-                    double lon = linkedPointSet.pointSet.getLon(point);
-                    if (GeometryUtils.containsPoint(egressService.serviceArea, lon, lat)) {
-                        filteredCosts.add(point);
-                        filteredCosts.add(cost + egressService.waitTimeSeconds);
```

Comment:
Consolidated in linkedPointSet.extendCostsToPoints (around https://github.com/conveyal/r5/pull/647/files#diff-dd60435552e26291d0b9b9a9ad8bfa02c92db82d5217698bc03f98665b96eebdR537)